cea/demand/building_properties.py

```diff
@@ -1019,3 +1017,21 @@ def get_thermal_resistance_surface(prop_envelope, weather_data):
 
     return thermal_resistance_surface_wall, thermal_resistance_surface_roof, thermal_resistance_surface_win
 
+
+def verify_hvac_system_combination(prop_hvac, prop_emission_cooling, prop_ventilation_system_and_control):
```

Comment:
Great idea!

I wonder if this could be made even more generaliz: perhaps there could be a boolean in the cooling tab of the HVAC database similar to the ones for the ventilation systems, e.g. `MECH_VENT`: for AS0-AS2 and AS5, it would be `FALSE` and for AS3-AS4 it would be `TRUE`. Then
```
needs_mech_vent = prop_emission_cooling[prop_emission_cooling['code'] == type_cs]['MECH_VENT'].values[0]
have_mech_vent = prop_ventilation_system_and_control[prop_ventilation_system_and_control['code'] == type_vent]['MECH_VENT'].values[0]
if (needs_mech_vent) & (not have_mech_vent):
[...]
```

Or am I just over-optimizing a solution that is already ok?

Reply:
we should use this logic to optimize the code and avoid the for loop!
inside the code, we could still write the array 'needs_mech_vent' based on 'type_cs', and compare 'needs_mech_vent' and 'have_mech_vent' for all buildings at the same time
would it work?